src/slider.ts

```diff
@@ -0,0 +1,33 @@
+import { HTML } from 'bigtest';
+
+export default HTML.extend('slider')
+  .selector('span[role="slider"]')
+  .filters({
+    ariaValueNow: (el) => {
+      return el.getAttribute('aria-valuenow')
+    },
+    dataIndex: (el) => el.getAttribute('data-index'),
+    index: (el) => {
+      const parent = el.parentNode;
+      const sliderNodes = parent?.querySelectorAll('span[role="slider"]') || [];
+
+      for (let i = 0; i < sliderNodes?.length; i++) {
+        if (el === sliderNodes[i]) {
+          return i;
+        }
+      }
+      return undefined;
+    }
+  })
+  .actions({
+    keyThumb: ({ perform }, key) => perform((el) => {
```

Comment:
@cowboyd @wKich We are going to have to handle Keyboard events and Mouse events maybe in this Interactor. Before I go down this this path is there a better way to handle this at the moment?

Reply:
In my opinion, it's better to think about a high-level API for an interactor. It doesn't matter how value could be changed by keyboard or by mouse. It's like you are trying to answer `what is the user's goal?` question. Does the user want to play around with a thumb, or does he want to know/set the slider's value? By saying `user`, I mean the end-user, who uses a web application with material-ui components.

For the slider, it could be like this:
```ts
.filters({
  minValue,
  value,
  maxValue,
})
.actions({
  setMin(),
  setValue(value),
  setMax()
})
```

We need to hide low-level details under the hood. So a developer can get nice solid interactors.